protocol/contracts/templegold/DaiGoldAuction.sol

```diff
@@ -0,0 +1,341 @@
+     * @param amount Amount to auction tokens
+     */
+    function recoverToken(
+        address token,
+        address to,
+        uint256 amount
+    ) external override onlyElevatedAccess {
+        if (to == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        if (amount == 0) { revert CommonEventsAndErrors.ExpectedNonZero(); }
+
+        if (token != address(templeGold)) {
+            emit CommonEventsAndErrors.TokenRecovered(to, token, amount);
+            IERC20(token).safeTransfer(to, amount);
+            return;
+        }
+
+        // auction started but cooldown pending
+        uint256 epochId = _currentEpochId;
+        EpochInfo storage info = epochs[epochId];
+        if (info.startTime == 0) { revert InvalidOperation(); }
+        if (info.isActive()) { revert AuctionActive(); }
+        if (info.hasEnded()) { revert AuctionEnded(); }
+        uint256 _totalAuctionTokenAmount = info.totalAuctionTokenAmount;
+        if (amount > _totalAuctionTokenAmount) { revert CommonEventsAndErrors.InvalidAmount(token, amount); }
+        /// @dev Epoch data is deleted and leftover amount is added to nextAuctionGoldAmount.
+        /// so admin should recover total auction amount for epoch if that's the requirement
+        delete epochs[epochId];
+        /// @dev `nextAuctionGoldAmount` is set to 0 in `startAuction`.
+        /// `nextAuctionGoldAmount > 0` if there was a distribution after `auctionStart` called
+        /// epoch is deleted. so if amount < totalAuctionTokenAmount for epoch, add leftover to next auction amount
+        unchecked {
+            nextAuctionGoldAmount += _totalAuctionTokenAmount - amount;
+        }
+
+        emit CommonEventsAndErrors.TokenRecovered(to, token, amount);
+        templeGold.safeTransfer(to, amount);
+    }
+
+    /**
+     * @notice Recover auction tokens for epoch with zero bids
+     * @param epochId Epoch Id
+     * @param to Recipient
+     */
+    function recoverTempleGoldForZeroBidAuction(uint256 epochId, address to) external override onlyElevatedAccess {
+        if (to == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+        // has to be valid epoch
+        if (epochId > _currentEpochId) { revert InvalidEpoch(); }
+        if (epochsWithoutBidsRecovered[epochId]) { revert AlreadyRecovered(); }
+        // epoch has to be ended
+        EpochInfo storage epochInfo = epochs[epochId];
+        if (!epochInfo.hasEnded()) { revert AuctionActive(); }
+        // bid token amount for epoch has to be 0
+        if (epochInfo.totalBidTokenAmount > 0) { revert InvalidOperation(); }
+
+        epochsWithoutBidsRecovered[epochId] = true;
+        uint256 amount = epochInfo.totalAuctionTokenAmount;
+        emit CommonEventsAndErrors.TokenRecovered(to, address(templeGold), amount);
+        templeGold.safeTransfer(to, amount);
+    }
+
+    /**
+     * @notice Mint and distribute TGOLD 
+     */
+    function distributeGold() external {
```

Comment:
Why is this function needed? I don't see any advantage over simply calling `TempleGold.mint()`

(just out of curiosity)

Reply:
It's a convenient function for `DaiGoldAuction` and `TempleGoldStaking` to enable distribution at any time on a standalone way, without switching conext to `TempleGold` to mint.